src/test/ui/fn/dyn-fn-alignment.rs

```diff
@@ -0,0 +1,24 @@
+// run-pass
+
+#![allow(dead_code)]
+#[repr(align(256))]
+struct A {
+    v: u8,
+}
+
+impl A {
+    fn f(&self) -> *const A {
+        assert_eq!(self as *const A as usize % 256, 0, "optimized out");
```

Comment:
The failure message here seems a bit odd?

Reply:
Yeah I've copied straight the test case from the issue but this is not needed for our tests. It's already covered in the main.